### ISSUE TITLE
`Could not find Cargo.toml in someday or any parent directory`

### ISSUE THREAD

**Issue by Ionizing** [neoclide/coc-rls#6]
# cod-rls has problem with finding `Cargo.toml`

When you `cargo new someroject` in a dir which is initialized to a `git` repo, like this:
```
ls -aR | grep ":" | sed -e 's/://' -e 's/[^-][^\/]*\//--/g' -e 's/^/   /' -e 's/-/|/'
   |-.git
   |---hooks
   |---info
   |---logs
   |-----refs
   |-------heads
   |-------remotes
   |---------origin
   |---objects
   |-----25
   |-----28
   |-----53
   |-----70
   |-----e5
   |-----info
   |-----pack
   |---refs
   |-----heads
   |-----remotes
   |-------origin
   |-----tags
   |-coctest
   |---src
```

and open `contest/src/main.rs`, `coc-rls` will inform that 
<img width="745" alt="image" src="https://user-images.githubusercontent.com/17506747/49693830-aaa08180-fbb8-11e8-8ac1-eb010bf3a7ba.png">

It seems that `coc-rls` has problem with reading the true WORKINGDIR.

**Comment by chemzqm** [comment]
> coc-rls has problem with reading the true WORKINGDIR.

It doesn't use cwd, it resolve `Cargo.toml` from current file.

I can't reproduce the problem.  Consider upload full output by use `  "tsserver.trace.server": "verbose",` in configuration:   "tsserver.trace.server": "verbose",

**Comment by Ionizing** [comment]
Sorry, I followed your step, but cannot get log as a result of "`tsserver.xxx command not found`".
But I tried to config `rls` in `coc-settings.json` as following:
```
        "rls": {
            "command": "rustup",
            "args": ["run", "nightly", "rls"],
            "filetypes": ["rust"],
            "settings": {},
            "enable": true,
            "trace.server": "verbose"
        }

```
The problem was reproduced, and I got the log here:

https://paste.ubuntu.com/p/8xvWPY8XDr/

**Comment by chemzqm** [comment]
Do you have `Cargo.toml` file in directory `file:///Users/ionizing/Documents/tests/rusttest/CLRS` ?
Maybe your `cargo` is just too old, and it can't generate the file for you, try upgrade.

**Comment by Ionizing** [comment]
No. `Cargo.toml` exists in `file:///Users/ionizing/Documents/tests/rusttest/CLRS/coctest`, and the rust project root directory is `file:///Users/ionizing/Documents/tests/rusttest/CLRS/coctest`. When I delete the `.git/` in `CLRS`, both `coc-rls` and `rls` works well.

**Comment by chemzqm** [comment]
I've fixed this issue in coc-rls, to fix the problem with configuration, you need to add `      "rootPatterns": ["Cargo.toml"],` to `rls` section.

**Comment by Ionizing** [comment]
I found a workaround: cd into the root dir of project, open src files via `NERDTree` and `Cargo.toml` can be found correctly.

**Comment by mike-kfed** [comment]
I ran into this today... This only happens when you have a git project with multiple rust crates inside, because the default rootPatterns look for the `.git` directory. Which makes it work without config by chance on "normal" rust crates, because they likely happen to have a `.git` dir at the same root as the `Cargo.toml`.

to make this work for sub-crates I opened the config of coc in vim with command `:CocConfig` and pasted this:
```json
{
	"languageserver": {
		"rls": {
			"command": "rustup",
			"args": ["run", "nightly", "rls"],
			"filetypes": ["rust"],
			"rootPatterns": ["Cargo.toml"]
		}
	}
}
```
this seems to be the minimal config to make it work. I don't fully understand how the coc-rls plugs into coc, but I'd vote for injecting the `"Cargo.toml"` into rootPatterns by default when the coc-rls is loaded. This way it actually looks for the cargo.toml in every case, and no custom config file writing is necessary (just having `languageserver -> rls -> rootPatterns` is not enough, all of the above is needed, probably causing other side-effects?)
